src/NServiceBus.NHibernate/Subscriptions/CachedSubscriptionPersister.cs

```diff
@@ -19,16 +19,14 @@ public CachedSubscriptionPersister(ISessionFactory sessionFactory, TimeSpan expi
 
         public override Task Subscribe(Subscriber subscriber, MessageType messageType, ContextBag context)
         {
-            base.Subscribe(subscriber, messageType, context);
             cache.Clear();
-            return Task.FromResult(0);
+            return base.Subscribe(subscriber, messageType, context);
```

Comment:
Not intentional but I figured it won't make things worse. Clearing the cache won't hurt much and subscribe failures should not happen that often in the stable system. I figures the code will be cleaner this way (vs using `await`).